src/core/geocms/qgsgeonoderequest.cpp

```diff
@@ -0,0 +1,405 @@
+      // WFS url : BASE_URI/geoserver/WORKSPACE/wfs
+      layerStruct.wfsURL = mBaseUrl + "/geoserver/" + layerWorkspace + "/wfs";
+      // XYZ url : set to empty string
+      layerStruct.xyzURL = "";
+
+      layers.append( layerStruct );
+    }
+  }
+  // Geonode version 2.7 or newer
+  else if ( ( majorVersion == 2 && minorVersion >= 7 ) || ( majorVersion >= 3 ) )
+  {
+    for ( int i = 0; i < layerList.count(); i++ )
+    {
+      LayerStruct layerStruct;
+      QVariantMap layer = layerList[i].toMap();
+      // Find WMS, WFS, and XYZ link
+      QVariantList layerLinks = layer["links"].toList();
+      layerStruct.wmsURL = QStringLiteral( "" );
+      layerStruct.wfsURL = QStringLiteral( "" );
+      layerStruct.xyzURL = QStringLiteral( "" );
+      for ( int j = 0; j < layerLinks.count(); j++ )
+      {
+        QVariantMap link = layerLinks[j].toMap();
+        if ( link.contains( "link_type" ) )
+        {
+          if ( link["link_type"] == "OGC:WMS" )
+          {
+            layerStruct.wmsURL = link["url"].toString();
+          }
+          if ( link["link_type"] == "OGC:WFS" )
+          {
+            layerStruct.wfsURL = link["url"].toString();
+          }
+          if ( link["link_type"] == "image" )
+          {
+            if ( link.contains( "name" ) && link["name"] == "Tiles" )
+            {
+              layerStruct.xyzURL = link["url"].toString();
+            }
+          }
+        }
+      }
+      if ( layer["typename"].toString().length() == 0 )
+      {
+        QStringList splitURL = layer["detail_url"].toString().split( "/" );
+        layerStruct.typeName = splitURL[ splitURL.length() - 1];
+      }
+      layerStruct.uuid = layer["uuid"].toString();
+      layerStruct.name = layer["name"].toString();
+      layerStruct.typeName = layer["typename"].toString();
+      layerStruct.title = layer["title"].toString();
+      layers.append( layerStruct );
+    }
+  }
+  return layers;
+}
+
+
+QStringList QgsGeoNodeRequest::serviceUrls( QString serviceType )
+{
+  QStringList *urls = new QStringList;
+  bool success = getLayers();
+
+  if ( !success )
+  {
+    return *urls;
+  }
+
+  QList<LayerStruct> layers = parseLayers( this->response() );
+
+  for ( int i = 0; i < layers.count(); i++ )
+  {
+    QString url;
+    if ( serviceType.toLower() == "wms" )
+    {
+      url = layers[i].wmsURL;
+    }
+    else if ( serviceType.toLower() == "wfs" )
+    {
+      url = layers[i].wfsURL;
+    }
+    else if ( serviceType.toLower() == "xyz" )
+    {
+      url = layers[i].xyzURL;
+    }
+    else
+    {
+      url = "";
+    }
+
+    if ( !url.contains( QLatin1String( "://" ) ) && url.length() > 0 )
+    {
+      url.prepend( "http://" );
+    }
+    if ( !urls->contains( url ) && url.length() > 0 )
+    {
+      urls->append( url );
+    }
+  }
+
+  return *urls;
+}
+
+
+QgsStringMap QgsGeoNodeRequest::serviceUrlData( QString serviceType )
+{
+  QgsStringMap *urls = new QgsStringMap;
```

Comment:
Hi @m-kuhn 
I moved the code to another method, so your comment is gone. I will check the outdated comment also to make sure we do not miss any comments.
Thanks